Integrations/integration-McAfeeNSM.yml

```diff
@@ -0,0 +1,679 @@
+        log('Showing {0}/{1} Alerts:'.format(res.retrievedAlertsCount, res.totalAlertsCount));
+        return alerts_to_entry('Showing {0}/{1} Alerts:'.format(res.retrievedAlertsCount, res.totalAlertsCount), res.alertsList);
+    }
+
+    function update_alerts(state, time_period, start_time, end_time, search, filter, new_state, new_assignee) {
+        var cmd_url = all_alerts_url(state, time_period, start_time, end_time, search, filter);
+        if ((new_state === undefined) && (new_assignee === undefined)) {
+            return 'Error! You must specify a new alert state or a new assignee';
+        }
+
+        var query = {};
+        if (new_state !== undefined) {
+            query.alertState = new_state;
+        }
+        if (new_assignee !== undefined) {
+            query.assignTo = new_assignee;
+        }
+        query = JSON.stringify(query);
+
+        var res = request('PUT', cmd_url, {}, query);
+        if (res.status === -1) {
+            return 'Error! Failed to update alerts.';
+        }
+
+        return get_alerts(state, time_period, start_time, end_time);
+    }
+
+    function get_alert_details(alert_id) {
+        var cmd_url = 'alerts/{0}'.format(alert_id);
+        var res = request('GET', cmd_url, {}, '');
+        return alert_to_entry('Alert: {0}'.format(res.name), res)
+    }
+
+    function get_ntba_monitors() {
+        var cmd_url = 'ntbamonitors';
+        var res = request('GET', cmd_url, {}, '');
+
+        return res;
+    }
+    function get_events(nba_id, hash, duration) {
+        var cmd_url = '{0}/endpointintelligence/{1}/events?duration={2}'.format(nba_id, hash, duration);
+        var res = request('GET', cmd_url, {}, '');
+
+        return res.eventList;
+    }
+
+    login();
+    // The command input arg holds the command sent from the user.
+    switch (command) {
+        case 'nsm-get-pcap':
+            return get_pcap_by_sensor(args.sensorID);
+        case 'nsm-get-sensors':
+            return get_sensors(args.domainID);
+
+        case 'nsm-get-domains':
+            return get_domains(args.domain);
+
+        case 'nsm-get-alerts':
+            return get_alerts(args.state, args.time_period, args.start_time, args.end_time, args.search, args.filter);
+        case 'nsm-get-alert-details':
+            return get_alert_details(args.alert_id);
+        case 'nsm-update-alerts':
+            return update_alerts(args.state, args.time_period, args.start_time, args.end_time, args.search, args.filter, args.new_state, args.new_assignee);
+
+        case 'nsm-get-attacks':
+            return get_attacks(args.attack_id);
+
+        case 'nsm-get-ips-policies':
+            return get_IPS_policies(args.domain_id);
+        case 'nsm-get-ips-policy-details':
+            return get_IPS_policy_details(args.policy_id);
+
+
+        // This is the call made when pressing the integration test button.
+        case 'test-module':
+            return 'ok';
```

Comment:
the test does not do anything and it should

usually, we call some general command that supposes to work always... maybe `get_domains()`?

Reply:
I think this is ok, as it just checks login (which is the most basic action)

Reply:
in this case, the login is tested, it is executing before the switch.
i can add the get domain as well